### DISCUSSION TITLE
Change html's lang attribute dynamically

### DISCUSSION THREAD

**Discussion by pmarxbraun** [gatsbyjs/gatsby#37042] [Help]
Hi,

I'm considering moving from `gatsby-plugin-react-helmet` to the new the built-in `Gatsby Head API` for my SEO comp.
However docs [tells to use](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-head/#editing-html-and-body)  onRenderBody to edit <html> tag and NOT the built-in Head.

```
exports.onRenderBody = ({ setHtmlAttributes }) => {
  setHtmlAttributes({ lang: "en" })
}
````

Is there a way to dynamically change this lang attribute ?
I got a multilang site and my `lang` for each page comes from a query as follow

``` 
const Home = ({ location, data }) => {
  const lang = data.site.siteMetadata.langDE

  return <HomeTemplate lang={lang} location={location} />
}

export default Home

export const query = graphql`
  query {
    site {
      siteMetadata {
        langDE
      }
    }
  }
`
```

Right now `HomeTemplate` is using `lang` in a SEO comp using `gatsby-plugin-react-helmet` so it works as helmet can mess around with `<html>`. But with `Gatsby Head API` I'm struggling to find a solution.

Thanks for your help

**Comment by byronwilliams** [comment]
This is referenced in multiple locations:

- https://github.com/gatsbyjs/gatsby/pull/36354 - can't be merged because the function `loadPageDataSync` doesn't actually exist
- https://github.com/gatsbyjs/gatsby/discussions/35841#discussioncomment-3357209
- https://github.com/gatsbyjs/gatsby/issues/36675#issuecomment-1264607615 - so you can probably get it to work if you do something like this on v4.24.0+

Try this:

```ts
import { GatsbySSR } from "gatsby";
import type PageContext from "../wherever"

export const onRenderBody: GatsbySSR["onRenderBody"] = (props) => {
  // only exists when NODE_ENV === "production" in v4.24.0
  if (props.loadPageDataSync) {
    const { loadPageDataSync, setHtmlAttributes, pathname } = props;
    const { pageContext } = loadPageDataSync(pathname).result as Record<
      string,
      PageContext
    >;

    setHtmlAttributes({ lang: pageContext.locale });
  }
};
```
